starfyre/parser.py

```diff
@@ -166,11 +170,25 @@ def handle_endtag(self, tag):
         if endtag_node.tag != "style" and endtag_node.tag != "script":
             if len(self.stack) > 0:
                 parent_node = self.stack[-1]      #this is last item/"top element" of stack
-                if parent_node.original_name == endtag_node.original_name:
-                    parent_node.children.extend(self.current_children)
-                    self.current_children = []
-                else:
-                    self.current_children.append(endtag_node)
+                if endtag_node.original_name != "": #this node is for "custom" component, so we should check for <slot> tags 
+                    #process endtag_node children
+                    new_children = []
+                    for child_component in endtag_node.children:
+                        if child_component.tag == "slot":
+                            new_children.extend(self.current_children)
+                        else:
+                            new_children.append(child_component)
+                    endtag_node.children = new_children   
+                    self.current_children = []  
+
+                
+                # if parent_node.original_name == endtag_node.original_name: #SK Why we need this condition? It is empty string by default
```

Comment:
we need this condition for name resolution. i.e.

let's say
```
a = <div></div>

```

And we have a code

```
<A><div></div></A>
```

The div tag will interfere with the original `A`.